src/dialect/mod.rs

```diff
@@ -636,6 +636,12 @@ pub trait Dialect: Debug + Any {
         false
     }
 
+    /// Returns true if this dialect allows dollar quoted strings
+    /// e.g. `SELECT $$Hello, world!$$` or `SELECT $tag$Hello, world!$tag$`
+    fn supports_dollar_quoted_string(&self) -> bool {
```

Comment:
I wondering would it make more sense to flip the condition to e.g. `supports_dollar_placeholder()`, setting that to true for sqlite and default false for the others? Thinking that might be the more conservative change of the two given that its the placeholder behavior that is more subtle/specific, (most of the other dialects technically don't support dollar quoted string so its less misleading if they don't have to explicitly flag otherwise)